lib/src/facts/resolvers/zpool_resolver.cc

```diff
@@ -30,25 +31,59 @@ namespace facter { namespace facts { namespace resolvers {
         if (!data.version.empty()) {
             facts.add(fact::zpool_version, make_value<string_value>(move(data.version)));
         }
-        if (!data.features.empty()) {
-            facts.add(fact::zpool_featurenumbers, make_value<string_value>(boost::join(data.features, ",")));
+        if (!data.feature_flags.empty()) {
+            facts.add(fact::zpool_featureflags, make_value<string_value>(boost::join(data.feature_flags, ",")));
+        }
+        if (!data.versions.empty()) {
+            facts.add(fact::zpool_versionnumbers, make_value<string_value>(boost::join(data.versions, ",")));
         }
     }
 
     zpool_resolver::data zpool_resolver::collect_data(collection& facts)
     {
         data result;
 
+        enum { UNKNOWN, FEATURES, VERSIONS } state = UNKNOWN;
+
         // Get the zpool version and features
-        static boost::regex zpool_version("ZFS pool version (\\d+)[.]");
-        static boost::regex zpool_feature("^\\s*(\\d+)[ ]");
+        static boost::regex zpool_version("^This system is currently running ZFS pool version (\\d+)\\.$");
```

Comment:
On FreeBSD, they are currently not localized.  BTW, it looks like leatherman sets `LC_ALL=C` in the command environment, so this should not be an issue.